src/simple_http.rs

```diff
@@ -49,6 +50,52 @@ impl SimpleHttpTransport {
     pub fn builder() -> Builder {
         Builder::new()
     }
+
+    fn request<R>(&self, req: impl serde::Serialize) -> Result<R, Error>
+        where R: for<'a> serde::de::Deserialize<'a>
+    {
+        // Open connection
+        let request_deadline = Instant::now() + self.timeout;
+        let mut sock = TcpStream::connect_timeout(&self.addr, self.timeout)?;
+
+        // Send HTTP request
+        sock.write_all(format!("POST {} HTTP/1.1\r\n", self.path).as_bytes())?;
```

Comment:
Could you break this apart so that it doesn't use `format!`?

Reply:
ah yeah that makes sense